src/highlighting.c

```diff
@@ -1776,3 +1776,23 @@ gboolean highlighting_is_code_style(gint lexer, gint style)
 	return !(highlighting_is_comment_style(lexer, style) ||
 		highlighting_is_string_style(lexer, style));
 }
+
+
+gchar **highlighting_get_keywords(GeanyFiletypeID filetype_id)
+{
+	GString *str = g_string_sized_new(1000);
+	gchar **keywords;
+	gchar **keyword_str;
+
+	foreach_strv(keyword_str, style_sets[filetype_id].keywords)
+	{
+		g_string_append(str, *keyword_str);
+		g_string_append_c(str, ' ');
+	}
+
+	keywords = g_strsplit(str->str, " ", -1);
```

Comment:
:-D

Yeah, a comment is a good idea because it really looks silly when you expect the keywords list is already separated. I actually expected this myself when I was writing the patch and just returned keywords and was then surprised by the results.